firebase_admin/auth.py

```diff
@@ -94,10 +96,653 @@ def verify_id_token(id_token, app=None):
+
+    def __init__(self, app):
+        g_credential = app.credential.get_credential()
+        session = transport.requests.AuthorizedSession(g_credential)
+        version_header = 'Python/Admin/{0}'.format(firebase_admin.__version__)
+        session.headers.update({'X-Client-Version': version_header})
+        self._session = session
+
+    def get_user(self, uid):
+        """Gets the user data corresponding to the specified user ID."""
+        _Validator.validate_uid(uid)
+        try:
+            response = self._request('post', 'getAccountInfo', json={'localId' : [uid]})
+        except requests.exceptions.RequestException as error:
+            self._handle_http_error(
+                _UserManager._INTERNAL_ERROR, 'Failed to get user: {0}.'.format(uid), error)
+        else:
+            if not response or not response.get('users'):
+                raise AuthError(
+                    _UserManager._USER_NOT_FOUND_ERROR,
+                    'No user record found for the provided user ID: {0}'.format(uid))
+            return UserRecord(response['users'][0])
+
+    def get_user_by_email(self, email):
+        """Gets the user data corresponding to the specified user email."""
+        _Validator.validate_email(email)
+        try:
+            response = self._request('post', 'getAccountInfo', json={'email' : [email]})
+        except requests.exceptions.RequestException as error:
+            self._handle_http_error(
+                _UserManager._INTERNAL_ERROR,
+                'Failed to get user by email: {0}.'.format(email), error)
+        else:
+            if not response or not response.get('users'):
+                raise AuthError(
+                    _UserManager._USER_NOT_FOUND_ERROR,
+                    'No user record found for the provided email: {0}'.format(email))
+            return UserRecord(response['users'][0])
+
+    def get_user_by_phone_number(self, phone_number):
+        """Gets the user data corresponding to the specified phone number."""
+        _Validator.validate_phone(phone_number)
+        try:
+            response = self._request(
+                'post', 'getAccountInfo', json={'phoneNumber' : [phone_number]})
+        except requests.exceptions.RequestException as error:
+            self._handle_http_error(
+                _UserManager._INTERNAL_ERROR,
+                'Failed to get user by phone number: {0}.'.format(phone_number), error)
+        else:
+            if not response or not response.get('users'):
+                raise AuthError(
+                    _UserManager._USER_NOT_FOUND_ERROR,
+                    'No user record found for the provided phone number: {0}'.format(phone_number))
+            return UserRecord(response['users'][0])
+
+    def create_user(self, **kwargs):
+        """Creates a new user account with the specified properties."""
+        payload = {}
+        for key, value in _UserManager._CREATE_USER_FIELDS.items():
```

Comment:
I wonder if the kwargs validation/copying logic can be shared between create_user and update_user.